docs/tutorial.md

```diff
@@ -0,0 +1,295 @@
+# DNS-based discovery Tree Creation: Basic Tutorial
+
+## Background
+
+The `tree_creator` is a command line utility used to create and update [EIP-1459](https://eips.ethereum.org/EIPS/eip-1459) compliant Merkle trees.
+It takes as input a list of node records (in ENR text encoding) and link entries pointing to other trees.
+The `tree_creator` utility will keep track of the tree sequence number and increase it whenever the encoded links or ENR entries are updated.
+The root domain is configurable.
+The most useful output is a map of subdomains to TXT records that can easily be converted to a zone file and deployed to a DNS name server,
+from where the encoded lists of links and ENR can be retrieved by clients.
+
+## How to build and run
+
+To build and run `tree_creator` using its default configuration:
+
+```bash
+# Build `tree_creator` utility
+make creator
+
+# Run tree creator utility with default configuration
+./build/tree_creator
+```
+
+To initialise the tree with some entries when running the utility,
+specify `--enr-record` or `--link` command line options.
+Both arguments may be repeated as many times as necessary.
+
+```bash
```

Comment:
Just a random idea, but an alterantive way of writing this might be with a heredoc https://en.wikipedia.org/wiki/Here_document#Unix_shells - because the arguments are so long I imagine it'll be more convenient to specify in a file